### PR TITLE
Support TLS with vault for the server-acl-init job

### PR DESCRIPTION
**Pull request by ishustava**
Changes proposed in this PR:
- Support TLS with vault for the server-acl-init job
- depends on #881 

How I've tested this PR:
acceptance tests

How I expect reviewers to test this PR:
code review

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Review by thisisnotashwin — APPROVED** [review]
💥 this is excellent!!

**Review by kschoche — APPROVED** [review]
🔥

File: charts/consul/templates/server-acl-init-job.yaml [hunk 1]
```diff
@@ -33,12 +33,23 @@ spec:
         component: server-acl-init
       annotations:
         "consul.hashicorp.com/connect-inject": "false"
+        {{- if (and .Values.global.secretsBackend.vault.enabled .Values.global.tls.enabled) }}
+        "vault.hashicorp.com/agent-pre-populate-only": "true"
```
Comment (kschoche) [inline charts/consul/templates/server-acl-init-job.yaml:37]: Not your fault but I'm not sure why this is necessary ([vault api docs aren't super clear](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-hashicorp-com-agent-pre-populate)), what is this super secret annotation used for?

Comment (david-yu) [reply charts/consul/templates/server-acl-init-job.yaml:37]: This is currently the only way to attach the vault agent as a sidecar and reliably have the job complete with the sidecar.

**Pull request merged**